x-pack/metricbeat/module/awsfargate/task_stats/data.go

```diff
@@ -39,15 +39,40 @@ func createEvent(stats *Stats) mb.Event {
 	regionName, clusterName := getRegionAndClusterName(stats.Container.Labels)
 	e.RootFields = createRootFields(stats, regionName)
 	if clusterName != "" {
-		e.MetricSetFields.Put("cluster_name", clusterName)
+		_, err := e.MetricSetFields.Put("cluster_name", clusterName)
+		if err != nil {
+			_ = fmt.Errorf("error putting metric set field 'cluster_name': %w", err)
+		}
 	}
 
-	taskName := stats.Container.Labels[taskLabel]
+	taskName := stats.taskInfo.Family
 	if taskName != "" {
-		e.MetricSetFields.Put("task_name", taskName)
+		_, err := e.MetricSetFields.Put("task_name", taskName)
+		if err != nil {
+			_ = fmt.Errorf("error putting metric set field 'task_name': %w", err)
+		}
+	}
+
+	taskDesiredStatus := stats.taskInfo.TaskDesiredStatus
+	if taskDesiredStatus != "" {
+		_, err := e.MetricSetFields.Put("task_desired_status", taskDesiredStatus)
+		if err != nil {
+			_ = fmt.Errorf("error putting metric set field 'task_desired_status': %w", err)
+		}
 	}
 
-	e.MetricSetFields.Put("identifier", generateIdentifier(stats.Container.Name, stats.Container.DockerId))
+	taskKnownStatus := stats.taskInfo.TaskKnownStatus
+	if taskKnownStatus != "" {
+		_, err := e.MetricSetFields.Put("task_known_status", taskKnownStatus)
+		if err != nil {
+			_ = fmt.Errorf("error putting metric set field 'task_known_status': %w", err)
```

Comment:
Do we need to log this error (similar below)?

Reply:
Not necessarily, we can choose to completely ignore it with something like `_, _ = e.MetricSetFields.Put(.....)`. `golint-ci` would pass.

Reply:
Thanks @zmoog for the insights. I removed the explicit error logging in https://github.com/elastic/beats/pull/32342/commits/d1ed939cbd80ce455b154968758430d6693e4797

Reply:
Good point! Thanks @zmoog and @girodav !